diff --git a/node_modules/react-native-calendars/src/calendar-list/index.d.ts b/node_modules/react-native-calendars/src/calendar-list/index.d.ts
index ab7a6e9..1812d55 100644
--- a/node_modules/react-native-calendars/src/calendar-list/index.d.ts
+++ b/node_modules/react-native-calendars/src/calendar-list/index.d.ts
@@ -20,6 +20,7 @@ export interface CalendarListProps extends CalendarProps, Omit<FlatListProps<any
     showScrollIndicator?: boolean;
     /** Whether to animate the auto month scroll */
     animateScroll?: boolean;
+    renderPlaceholder?:(year: number, month: number) => React.ReactElement;
 }
 export interface CalendarListImperativeMethods {
     scrollToDay: (date: XDate | string, offset: number, animated: boolean) => void;
diff --git a/node_modules/react-native-calendars/src/calendar-list/index.js b/node_modules/react-native-calendars/src/calendar-list/index.js
index f8943de..86ad94d 100644
--- a/node_modules/react-native-calendars/src/calendar-list/index.js
+++ b/node_modules/react-native-calendars/src/calendar-list/index.js
@@ -40,7 +40,7 @@ const CalendarList = (props, ref) => {
     /** View props */
     testID, style: propsStyle, onLayout, removeClippedSubviews, 
     /** ScrollView props */
-    horizontal = false, pagingEnabled, scrollEnabled = true, nestedScrollEnabled = true, scrollsToTop = false, keyExtractor = (_, index) => String(index), keyboardShouldPersistTaps, onScrollBeginDrag, onScrollEndDrag, onMomentumScrollBegin, onMomentumScrollEnd, 
+    horizontal = false, pagingEnabled, scrollEnabled = true, nestedScrollEnabled = true, scrollsToTop = false, keyExtractor = (_, index) => String(index), keyboardShouldPersistTaps, onScrollBeginDrag, onScrollEndDrag, onMomentumScrollBegin, onMomentumScrollEnd, renderPlaceholder,
     /** FlatList props */
     onEndReachedThreshold, onEndReached } = props;
     const calendarProps = extractCalendarProps(props);
@@ -152,7 +152,7 @@ const CalendarList = (props, ref) => {
         const testId = `${testID}.item_${year}-${month}`;
         return (<CalendarListItem {...calendarProps} testID={testId} markedDates={getMarkedDatesForItem(item)} item={item} style={calendarStyle} 
         // @ts-expect-error - type mismatch - ScrollView's 'horizontal' is nullable
-        horizontal={horizontal} calendarWidth={calendarWidth} calendarHeight={calendarHeight} scrollToMonth={scrollToMonth} visible={isDateInRange(item)}/>);
+        horizontal={horizontal} calendarWidth={calendarWidth} calendarHeight={calendarHeight} scrollToMonth={scrollToMonth} visible={isDateInRange(item)} renderPlaceholder={renderPlaceholder}/>);
     }, [horizontal, calendarStyle, calendarWidth, testID, getMarkedDatesForItem, isDateInRange, calendarProps]);
     const renderStaticHeader = () => {
         if (staticHeader && horizontal) {
diff --git a/node_modules/react-native-calendars/src/calendar-list/item.d.ts b/node_modules/react-native-calendars/src/calendar-list/item.d.ts
index 1a8aca0..c642a79 100644
--- a/node_modules/react-native-calendars/src/calendar-list/item.d.ts
+++ b/node_modules/react-native-calendars/src/calendar-list/item.d.ts
@@ -3,13 +3,14 @@ import React from 'react';
 import { Theme } from '../types';
 import { CalendarProps } from '../calendar';
 export type CalendarListItemProps = CalendarProps & {
-    item: any;
+    item: XDate;
     calendarWidth?: number;
     calendarHeight?: number;
     horizontal?: boolean;
     theme?: Theme;
     scrollToMonth?: (date: XDate) => void;
     visible?: boolean;
+    renderPlaceholder?:(year: number, month: number) => React.ReactElement;
 };
 declare const CalendarListItem: React.MemoExoticComponent<(props: CalendarListItemProps) => JSX.Element>;
 export default CalendarListItem;
diff --git a/node_modules/react-native-calendars/src/calendar-list/item.js b/node_modules/react-native-calendars/src/calendar-list/item.js
index 0e5be76..a00eeb2 100644
--- a/node_modules/react-native-calendars/src/calendar-list/item.js
+++ b/node_modules/react-native-calendars/src/calendar-list/item.js
@@ -1,11 +1,11 @@
 import React, { useRef, useMemo, useCallback } from 'react';
-import { Text } from 'react-native';
+import { Text,View } from 'react-native';
 import { toMarkingFormat } from '../interface';
 import { extractCalendarProps } from '../componentUpdater';
 import styleConstructor from './style';
 import Calendar from '../calendar';
 const CalendarListItem = React.memo((props) => {
-    const { item, theme, scrollToMonth, horizontal, calendarHeight, calendarWidth, style: propsStyle, headerStyle, onPressArrowLeft, onPressArrowRight, visible } = props;
+    const { item, theme, scrollToMonth, horizontal, calendarHeight, calendarWidth, style: propsStyle, headerStyle, onPressArrowLeft, onPressArrowRight, visible,renderPlaceholder } = props;
     const style = useRef(styleConstructor(theme));
     const calendarProps = extractCalendarProps(props);
     const dateString = toMarkingFormat(item);
@@ -52,6 +52,12 @@ const CalendarListItem = React.memo((props) => {
         }
     }, [onPressArrowRight, scrollToMonth]);
     if (!visible) {
+        if(renderPlaceholder){
+            const year = item.getFullYear();
+            const month = item.getMonth();
+            return <View style={calendarStyle}>{renderPlaceholder(year,month)}</View>;
+            
+        }
         return (<Text style={textStyle}>{dateString}</Text>);
     }
     return (<Calendar hideArrows={true} hideExtraDays={true} {...calendarProps} current={dateString} style={calendarStyle} headerStyle={horizontal ? headerStyle : undefined} disableMonthChange onPressArrowLeft={horizontal ? _onPressArrowLeft : onPressArrowLeft} onPressArrowRight={horizontal ? _onPressArrowRight : onPressArrowRight}/>);
diff --git a/node_modules/react-native-calendars/src/calendar/day/period/index.js b/node_modules/react-native-calendars/src/calendar/day/period/index.js
index f27db0e..79c8afa 100644
--- a/node_modules/react-native-calendars/src/calendar/day/period/index.js
+++ b/node_modules/react-native-calendars/src/calendar/day/period/index.js
@@ -4,9 +4,12 @@ import { TouchableWithoutFeedback, TouchableOpacity, Text, View } from 'react-na
 import { xdateToData } from '../../../interface';
 import styleConstructor from './style';
 import Dot from '../dot';
+import XDate from 'xdate';
+import { backgroundColor } from 'react-native-calendars/src/style';
 const PeriodDay = (props) => {
     const { theme, marking, date, onPress, onLongPress, state, accessibilityLabel, testID, children } = props;
     const dateData = date ? xdateToData(date) : undefined;
+    const xdate = new XDate(date)
     const style = useRef(styleConstructor(theme));
     const markingStyle = useMemo(() => {
         const defaultStyle = { textStyle: {}, containerStyle: {} };
@@ -40,6 +43,7 @@ const PeriodDay = (props) => {
             }
             if (marking.customContainerStyle) {
                 defaultStyle.containerStyle = marking.customContainerStyle;
+             
             }
             return defaultStyle;
         }
@@ -60,7 +64,7 @@ const PeriodDay = (props) => {
             const start = markingStyle.startingDay;
             const end = markingStyle.endingDay;
             if (start && !end) {
-                containerStyle.push({ backgroundColor: markingStyle.startingDay?.backgroundColor });
+                containerStyle.push({ backgroundColor: markingStyle.startingDay?.backgroundColor,});
             }
             else if (end && !start || end && start) {
                 containerStyle.push({ backgroundColor: markingStyle.endingDay?.backgroundColor });
@@ -88,12 +92,13 @@ const PeriodDay = (props) => {
     }, [marking, state]);
     const fillerStyles = useMemo(() => {
         const leftFillerStyle = { backgroundColor: undefined };
-        const rightFillerStyle = { backgroundColor: undefined };
+        const rightFillerStyle = { backgroundColor: undefined ,borderRadius:undefined};
         let fillerStyle = {};
         const start = markingStyle.startingDay;
         const end = markingStyle.endingDay;
         if (start && !end) {
             rightFillerStyle.backgroundColor = markingStyle.startingDay?.backgroundColor;
+
         }
         else if (end && !start) {
             leftFillerStyle.backgroundColor = markingStyle.endingDay?.backgroundColor;
@@ -102,17 +107,34 @@ const PeriodDay = (props) => {
             leftFillerStyle.backgroundColor = markingStyle.day?.backgroundColor;
             rightFillerStyle.backgroundColor = markingStyle.day?.backgroundColor;
             fillerStyle = { backgroundColor: markingStyle.day?.backgroundColor };
+       
         }
         return { leftFillerStyle, rightFillerStyle, fillerStyle };
     }, [marking]);
-    const renderFillers = () => {
-        if (marking) {
+    const renderFillers = (xdate) => {
+        const start = markingStyle.startingDay;
+        const end = markingStyle.endingDay;
+        //todo color theme으로 유동성있게 관리하기
+        const lrColor = '#EFF6FE'
+        
+        if(xdate.getDay()===0){
+            return (<View style={[style.current.fillers, fillerStyles.fillerStyle,{borderTopLeftRadius:30,borderBottomLeftRadius:30}]}>
+                <View style={[style.current.leftFiller, fillerStyles.leftFillerStyle,{borderTopLeftRadius:30,borderBottomLeftRadius:30},end? {backgroundColor:lrColor}:{borderRadius:0}]}/>
+                <View style={[style.current.rightFiller, fillerStyles.rightFillerStyle,start? {backgroundColor:lrColor}:{borderRadius:0}]}/>
+              </View>);
+        }else if(xdate.getDay() === 6){
+            return(<View style={[style.current.fillers, fillerStyles.fillerStyle,{borderTopRightRadius:30,borderBottomRightRadius:30}]}>
+                <View style={[style.current.leftFiller, fillerStyles.leftFillerStyle,end? {backgroundColor:lrColor}:{borderRadius:0}]}/>
+                <View style={[style.current.rightFiller, fillerStyles.rightFillerStyle,{borderTopRightRadius:30,borderBottomRightRadius:30},start? {backgroundColor:lrColor}:{borderRadius:0}]}/>
+              </View>);
+        }else  {
             return (<View style={[style.current.fillers, fillerStyles.fillerStyle]}>
-          <View style={[style.current.leftFiller, fillerStyles.leftFillerStyle]}/>
-          <View style={[style.current.rightFiller, fillerStyles.rightFillerStyle]}/>
+          <View style={[style.current.leftFiller, fillerStyles.leftFillerStyle,end? {backgroundColor:lrColor}:{borderRadius:0}]}/>
+          <View style={[style.current.rightFiller, fillerStyles.rightFillerStyle,start? {backgroundColor:lrColor}:{borderRadius:0}]}/>
         </View>);
         }
     };
+
     const _onPress = useCallback(() => {
         onPress?.(dateData);
     }, [onPress]);
@@ -122,7 +144,7 @@ const PeriodDay = (props) => {
     const Component = marking ? TouchableWithoutFeedback : TouchableOpacity;
     return (<Component testID={testID} onPress={_onPress} onLongPress={_onLongPress} disabled={marking?.disableTouchEvent} accessible accessibilityRole={marking?.disableTouchEvent ? undefined : 'button'} accessibilityLabel={accessibilityLabel}>
       <View style={style.current.wrapper}>
-        {renderFillers()}
+        {renderFillers(xdate)}
         <View style={containerStyle}>
           <Text allowFontScaling={false} style={textStyle}>
             {String(children)}
diff --git a/node_modules/react-native-calendars/src/types.d.ts b/node_modules/react-native-calendars/src/types.d.ts
index 31103ce..a80d83f 100644
--- a/node_modules/react-native-calendars/src/types.d.ts
+++ b/node_modules/react-native-calendars/src/types.d.ts
@@ -74,6 +74,8 @@ export interface Theme {
     arrowHeight?: number;
     arrowWidth?: number;
     weekVerticalMargin?: number;
+    'stylesheet.calendar.header'?:object;
+    'stylesheet.day.period'?:object;
     stylesheet?: {
         calendar?: {
             main?: object;
